src/lua/init.lua

```diff
@@ -194,6 +194,43 @@ local function run_preload()
     end
 end
 
+--
+-- Return true if module with given `path` is required to use box.error
+-- to indicate error and error trace should be set the module function
+-- invocation place.
+--
+local function trace_check_is_required(path)
+    local modules = {
+        'builtin/box/schema.lua',
+        'builtin/box/session.lua',
+        'builtin/digest.lua',
+        'builtin/error.lua',
+        'builtin/tarantool.lua',
+    }
+    for _, m in ipairs(modules) do
+        if path == m then
+            return true
```

Comment:
- Let's store the table in a global variable.
- Let's make module names keys in the table to speed up lookup.

Reply:
Done.

```diff
diff --git a/src/lua/init.lua b/src/lua/init.lua
index ca4910e57..70ddd7515 100644
--- a/src/lua/init.lua
+++ b/src/lua/init.lua
@@ -194,25 +194,25 @@ local function run_preload()
     end
 end
 
+--
+-- List of modules (by path) which are required to use box.error and
+-- set trace properly.
+--
+local trace_check_required_modules = {
+    ['builtin/box/schema.lua'] = true,
+    ['builtin/box/session.lua'] = true,
+    ['builtin/digest.lua'] = true,
+    ['builtin/error.lua'] = true,
+    ['builtin/tarantool.lua']= true,
+}
+
 --
 -- Return true if module with given `path` is required to use box.error
 -- to indicate error and error trace should be set the module function
 -- invocation place.
 --
 local function trace_check_is_required(path)
-    local modules = {
-        'builtin/box/schema.lua',
-        'builtin/box/session.lua',
-        'builtin/digest.lua',
-        'builtin/error.lua',
-        'builtin/tarantool.lua',
-    }
-    for _, m in ipairs(modules) do
-        if path == m then
-            return true
-        end
-    end
-    return false
+    return trace_check_required_modules[path] ~= nil
 end
 
 --

```